testsuite/openshift/kuadrant.py

```diff
@@ -1,29 +1,87 @@
 """Kuadrant CR object"""
 
-from openshift_client import selector
+import dataclasses
+from functools import cached_property
 
-from testsuite.openshift import OpenShiftObject, modify
+from openshift_client import selector, timeout
+
+from testsuite.openshift import CRObject
 from testsuite.openshift.deployment import Deployment
 
 
-class KuadrantCR(OpenShiftObject):
+class _CRObject:
+    """
+    Base class for Kuadrant sub components:
+        Authorino - spec.authorino
+        Limitador - spec.limitador
+    """
+
+    def __init__(self, kuadrant_cr, spec_name):
+        super().__init__()
+        self.kuadrant_cr = kuadrant_cr
+        self.spec_name = spec_name
+
+    @property
+    def committed(self):
+        """Reimplementation of commit from OpenshiftObject. Returns committed from KuadrantCR"""
+        return self.kuadrant_cr.committed
+
+    def safe_apply(self):
+        """Reimplementation of modify_and_apply from OpenshiftObject"""
+
+        result, status = self.kuadrant_cr.modify_and_apply(lambda _: True, retries=2)
+        assert status
+        return result
+
+    def __getitem__(self, name):
+        return self.kuadrant_cr.model.spec[self.spec_name][name]
+
+    def __setitem__(self, name, value):
+        if dataclasses.is_dataclass(value):
+            self.kuadrant_cr.model.spec[self.spec_name][name] = dataclasses.asdict(value)
+        else:
+            self.kuadrant_cr.model.spec[self.spec_name][name] = value
+
+    def wait_for_ready(self):
+        """Waits until Kuadrant CR reports ready status"""
+        self.kuadrant_cr.wait_for_ready()
+
+
+class _Authorino(_CRObject):
+    """Authorino-like object from KuadrantCR `spec.authorino`"""
+
+
+class _Limitador(_CRObject):
+    """Limitador-like object from KuadrantCR `spec.limitador`"""
+
+
+class KuadrantCR(CRObject):
     """Represents Kuadrant CR objects"""
 
     LIMITADOR = "limitador-limitador"
 
-    @property
-    def limitador(self) -> dict:
-        """Returns spec.limitador from Kuadrant object"""
-        return self.model.spec.setdefault("limitador", {})
+    @cached_property
```

Comment:
Wont hurt but not sure if needed, there is not state saved in the class and it is inexpensive operation